src/syscalls/main.rs

```diff
@@ -0,0 +1,293 @@
+            let pid: u32 = matches
+            .value_of("pid")
+            .unwrap()
+            .parse()
+            .expect("Invalid pid");
+        
+        code = format!("#define FILTER_PID {} \n {}", pid, code);
+    }
+    if matches.is_present("failures") {
+        code = format!("#define FILTER_FAILED {}", code);
+    }
+    if matches.is_present("errorno") {
+        let errorno = matches.value_of("errorno").unwrap();
+        code = format!("#define FILTER_ERRNO {} \n {}", errorno, code)
+    }
+    if matches.is_present("latency") {
+        code = format!("#define LATENCY \n {}", code);
+    }
+
+    code
+}
+
+fn do_main(runnable: Arc<AtomicBool>) -> Result<(), BccError> {
+    let matches = App::new("syscalls")
+        .about("Summarize number of syscalls and their latencies")
+        .arg(
+            Arg::with_name("interval")
+                .long("interval")
+                .value_name("Seconds")
+                .help("Integration window duration and period for stats output")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("windows")
+                .long("windows")
+                .value_name("Count")
+                .help("The number of intervals before exit")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("milliseconds")
+                .long("milliseconds")
+                .short("M")
+                .help("Display the timestamps in milliseconds")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("microseconds")
+                .long("microseconds")
+                .short("m")
+                .help("Display the timestamps in microseconds")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("pid")
+                .long("pid")
+                .short("p")
+                .help("Trace only the given pid")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("top")
+                .long("top")
+                .short("t")
+                .help("Display the top syscalls by count or latency")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("errorno")
+                .long("errorno")
+                .short("e")
+                .help("Trace only the syscalls that return the given error (numeric or EPERM, ...)")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("failures")
+                .long("failures")
+                .short("x")
+                .help("Trace only failed syscalls")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("latency")
+                .long("latency")
+                .short("L")
+                .help("Collect syscall latency")
+                .takes_value(false),
+        )
+        .get_matches();
+
+    let interval: usize = matches
```

Comment:
It might be better here to have the default value as the same type, this way we can set it as a const at the top of the file, or keep it in a common module. Maybe map(parse) the argument first so the unwrap_or can have a usize?